src/runtime/Converter.cs

```diff
@@ -326,6 +326,20 @@ internal static NewReference ToPythonImplicit(object? value)
                     throw new ArgumentException("We should never receive instances of other managed types");
             }
 
+            // NOTE: generate a delegate to cast a python callable into a System.Delegate
```

Comment:
This should either be part of `ToPrimitive`, or at very least moved down so that codecs could override this behavior. 

Also @filmor even if this is corrected this is a breaking change as it might fail or change overload resolution.

Reply:
`void Foo(PyObject x)` vs `void Foo(System.Action x)` previously would unambiguously resolve to the first variant. With this change it will either fail due to non-unique resolution or attempt to prioritize, which might or might not result in the second one being selected.